src/main/java/gregtech/api/recipes/ingredients/GTRecipeItemInput.java

```diff
@@ -0,0 +1,163 @@
+package gregtech.api.recipes.ingredients;
+
+import gregtech.api.recipes.ingredients.NBTMatching.NBTMatcher;
+import gregtech.api.recipes.ingredients.NBTMatching.NBTcondition;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraftforge.fluids.FluidStack;
+
+import java.lang.ref.WeakReference;
+import java.util.Objects;
+
+public class GTRecipeItemInput implements IGTRecipeInput {
+    int amount;
+    boolean isConsumable = true;
+    NBTMatcher nbtMatcher;
+    NBTcondition nbtCondition;
+    ItemStack inputStacks;
+
+    protected GTRecipeItemInput(ItemStack stack) {
+        this.inputStacks = stack;
+        amount = stack.getCount();
+    }
+
+    protected GTRecipeItemInput(ItemStack stack, int amount) {
+        this.inputStacks = stack;
+        this.amount = amount;
+    }
+
+    public static GTRecipeItemInput getOrCreate(ItemStack stack, int amount) {
+        stack.setCount(amount);
```

Comment:
Do we want to copy the stack first before we change the count here? Or instead just call `new GTRecipeItemInput(Itemstack, int)` instead of the Itemstack only version.